CMakeLists.txt

```diff
@@ -41,6 +41,12 @@ else()
     set(NANOARROW_NAMESPACE_DEFINE "// #define NANOARROW_NAMESPACE YourNamespaceHere")
 endif()
 
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+    set(NANOARROW_DEBUG_DEFINE "#define NANOARROW_DEBUG 1")
+else()
+    set(NANOARROW_DEBUG_DEFINE "// #define NANOARROW_DEBUG 1")
+endif()
```

Comment:
FWIW, CMake has an explicit feature for this: [target_compile_definitions](https://cmake.org/cmake/help/latest/command/target_compile_definitions.html)

Reply:
Ah, though I suppose this is different since it'll propagate to anyone who includes nanoarrow_config.h.

Reply:
We may want to use generator expression: https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html

```cmake
# We don't need to put this under if(CMAKE_BUILD_TYPE STREQUAL "Debug")
target_compile_definitions(nanoarrow PUBLIC "$<$<CONFIG:Debug>:NANOARROW_DEBUG>")
```